File: app/src/main/java/lt/tesonet/network/ApiService.java
package lt.tesonet.network;

import java.util.List;

import io.reactivex.Single;
import lt.tesonet.model.Server;
import lt.tesonet.model.User;
import lt.tesonet.model.Token;
import retrofit2.Response;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface ApiService {

    @POST("tokens")
    @Headers("Content-Type: application/json")
    Single<Response<Token>> logIn(@Body User user);

    @GET("servers")
    Single<Response<List<Server>>> getServers(@Header("Authorization") String header);
}

File: app/src/main/java/com/jonastiskus/testio/network/service/ApiService.kt
package com.jonastiskus.testio.network.service

import com.jonastiskus.testio.model.AuthGsonModel
import com.jonastiskus.testio.model.ServerGsonModel
import retrofit2.Call
import retrofit2.http.*

interface ApiService {

    @POST("tokens")
    @Headers("Content-type: application/json")
    fun getToken(@Query("username")userName : String,
                 @Query("password") password : String) : Call<AuthGsonModel>

    @GET("servers")
    @Headers("Content-type: application/json")
    fun getServers(@Header("Authorization") authToken : String) : Call<List<ServerGsonModel>>

}